File: test-docai.js
const { DocumentProcessorServiceClient } = require('@google-cloud/documentai');
const fs = require('fs');

async function runTest() {
  const keyFilePath = '/home/marcel/documents-ai_backup/keys/service-account.json';
  
  if (!fs.existsSync(keyFilePath)) {
    console.error(`❌ Datei nicht gefunden: ${keyFilePath}`);
    return;
  }

  const credentials = JSON.parse(fs.readFileSync(keyFilePath, 'utf8'));

  // Daten aus deinen erfolgreichen Python-Lauf
  const projectId = 'gen-lang-client-0151377967'; 
  const projectNumber = '841608965746';           
  const location = 'eu';                           
  const processorId = '7e5c6ef084f5281';           

  console.log('--- Node.js Regionaler Test (EU) ---');

  // 1. Client-Initialisierung
  const client = new DocumentProcessorServiceClient({
    apiEndpoint: 'eu-documentai.googleapis.com',
    credentials,
    projectId: projectId,
  });

  // 2. Pfad-Konstruktion
  const name = `projects/${projectNumber}/locations/${location}/processors/${processorId}`;

  try {
    console.log(`⏳ Rufe Prozessor ab via: ${name}`);

    /**
     * DER ENTSCHEIDENDE FIX:
     * Wir übergeben den servicePath als zweites Argument direkt im Aufruf.
     * Das ist das Node-Äquivalent zu den Python 'ClientOptions'.
     */
    const [processor] = await client.getProcessor(
      { name }, 
      { servicePath: 'eu-documentai.googleapis.com' } 
    );

    console.log('✅ ERFOLG!');
    console.log(`Anzeige-Name: ${processor.displayName}`);
    console.log(`Status: ${processor.state}`);

  } catch (err) {
    console.error('❌ FEHLER:');
    console.error(err.message);
    
    if (err.message.includes('documentai.googleapis.com') && !err.message.includes('eu-')) {
        console.log('\n💡 Hinweis: Das SDK hat den eu-Endpoint trotz Anweisung ignoriert.');
    }
  }
}

runTest();

File: test_docai.py
import sys
import os
from google.cloud import documentai_v1 as documentai
from google.api_core.client_options import ClientOptions

def test_connection():
    # Daten exakt aus deinen Bildern
    PROJECT_NUMBER = "841608965746"
    LOCATION = "eu"
    PROCESSOR_ID = "7e5c6ef084f5281"
    
    if len(sys.argv) < 2:
        print("❌ Fehler: Bitte Pfad zur service-account.json angeben.")
        return
    
    key_file = sys.argv[1]
    # Setze die Umgebungsvariable für die Authentifizierung
    os.environ["GOOGLE_APPLICATION_CREDENTIALS"] = key_file

    print(f"--- Starte Regionaler Test (EU) ---")
    
    try:
        # Erpringe den regionalen Endpunkt
        options = ClientOptions(api_endpoint=f"{LOCATION}-documentai.googleapis.com")
        
        # Erstelle den Client mit den expliziten Optionen
        client = documentai.DocumentProcessorServiceClient(client_options=options)
        
        # Baue den vollständigen Ressourcennamen
        resource_name = client.processor_path(PROJECT_NUMBER, LOCATION, PROCESSOR_ID)
        
        print(f"⏳ Versuche Zugriff auf: {resource_name}")
        
        # Teste den Zugriff durch Abruf der Prozessor-Metadaten
        processor = client.get_processor(name=resource_name)
        
        print(f"✅ ERFOLG!")
        print(f"Name: {processor.display_name}")
        print(f"Status: {processor.state.name}")

    except Exception as e:
        print(f"❌ FEHLER: {e}")

if __name__ == "__main__":
    test_connection()
